File: heap/506_relative_ranks.py
class Solution:
    def findRelativeRanks(self, score: List[int]) -> List[str]:
        maxHeap = []
        for idx, val in enumerate(score):
            heapq.heappush(maxHeap, (-val, idx))

        place = 1
        res = [''] * len(score)
        while maxHeap:
            pos = heapq.heappop(maxHeap)[1]
            if place > 3:
                rank = str(place)
            elif place == 1:
                rank = 'Gold Medal'
            elif place == 2:
                rank = 'Silver Medal'
            elif place == 3:
                rank = 'Bronze Medal'
            
            res[pos] = rank
            place += 1
            
        return res

        

File: array/506_relative_ranks.java
class Solution {
    public String[] findRelativeRanks(int[] score) {
        int n = score.length;
        int[] copy = new int[n];
        System.arraycopy(score, 0, copy, 0, n);

        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < n; i++) {
            map.put(copy[i], i);
        }

        Arrays.sort(copy);

        String[] res = new String[n];
        for (int i = 0; i < n; i++) {
            if (i == 0) {
                res[map.get(copy[n - i - 1])] = "Gold Medal";
            } else if (i == 1) {
                res[map.get(copy[n - i - 1])] = "Silver Medal";
            } else if (i == 2) {
                res[map.get(copy[n - i - 1])] = "Bronze Medal";                
            } else {
                res[map.get(copy[n - i - 1])] = Integer.toString(i+1);
            }
        }
        return res;
    }
}